### PR TITLE
Panel onOpen and onOpened

### LINKED ISSUE #8562: Add some kind of opened / transitionend event to layer / panel / ...
**Describe the feature that you would like added**
I'd like to have an event, that fires in case a Layer (for example of a Panel) has finished its opening transition and is visible in a stable state (e.g. *onLayerOpened*). The reason is, that opening a Panel with some complex content opens with long delay and choppy in some older browsers (actually it's IE11 ...), because rendering runs in parallel with the transition.
So in order to improve user experience I'd like to open the empty Panel first and then start rendering the child components. This way the user gets instant feedback to his click and the application feels more responsive.

**What component or utility would this be added to**
Current demand comes from the Panel component, but actually this could be used wherever an animated Layer is involved.

**Have you discussed this feature with our team, and if so, who**
No.

**Additional context/screenshots**
If further information is needed, feel free to ask me :)

### PR DESCRIPTION
**Pull request by kkjeer**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #8562 
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Adds `onOpen` (called when the Panel opens, before animation completes) and `onOpened` (called when the Panel opens, after animation completes) callback properties to Panel, analogous to the existing `onDismiss` and `onDismissed`.

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/8623)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Panel/Panel.test.tsx [hunk 1]
```diff
@@ -25,6 +25,43 @@ describe('Panel', () => {
...
+      expect(openedCalled).toEqual(false);
+
+      setTimeout(() => {
+        try {
+          expect(openedCalled).toEqual(true);
+          done();
+        } catch (e) {
+          done(e);
+        }
+      }, 250);
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/Panel/Panel.test.tsx]: I am assuming the 250ms here are to wait for the animation of opening the panel to finish. Maybe in this case use `jest.useFakeTimers()` like here:
https://github.com/OfficeDev/office-ui-fabric-react/blob/8ffd8134f6ebbf9ee0785e559e28b146d9de2edf/packages/office-ui-fabric-react/src/components/TextField/TextField.test.tsx#L332

Comment (dzearing) [reply packages/office-ui-fabric-react/src/components/Panel/Panel.test.tsx]: @Vitalius1 wouldn't that only apply to setTimeouts and intervals? Does that affect the transition events at all? https://jestjs.io/docs/en/timer-mocks

Comment (dzearing) [reply packages/office-ui-fabric-react/src/components/Panel/Panel.test.tsx]: Yes that would work.

Comment (dzearing) [reply packages/office-ui-fabric-react/src/components/Panel/Panel.test.tsx]: I just used it over the weekend; it's pretty great, except that it doesn't seem to work well with clearTimeout. (it still seems to execute the callback.) If you hit that, sinon has the same helper and it works better with clearTimeout scenarios. But I think the `jest.useFakeTimers` helper will work here.

**Comment by kkjeer** [comment]
@msft-github-bot make sure @dzearing approves

**Comment by kkjeer** [comment]
@msft-github-bot forget everything I just told you

**Pull request merged**